client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogmaRequestPreparation.java

```diff
@@ -0,0 +1,295 @@
+     * {@link CentralDogma#getHistory(String, String, Revision, Revision, String)} request to the
+     * Central Dogma repository.
+     */
+    public CentralDogmaHistoryRequest history() {
+        return new CentralDogmaHistoryRequest(this, "/**");
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaHistoryRequest} that is used to send
+     * {@link CentralDogma#getHistory(String, String, Revision, Revision, String)} request to the
+     * Central Dogma repository.
+     * A path pattern is a variant of glob:
+     * <ul>
+     *   <li>{@code "/**"} - find all files recursively</li>
+     *   <li>{@code "*.json"} - find all JSON files recursively</li>
+     *   <li>{@code "/foo/*.json"} - find all JSON files under the directory {@code /foo}</li>
+     *   <li><code>"/&#42;/foo.txt"</code> - find all files named {@code foo.txt} at the second depth level</li>
+     *   <li>{@code "*.json,/bar/*.txt"} - use comma to specify more than one pattern. A file will be matched
+     *                                     if <em>any</em> pattern matches.</li>
+     * </ul>
+     */
+    public CentralDogmaHistoryRequest history(String pathPattern) {
+        requireNonNull(pathPattern, "pathPattern");
+        return new CentralDogmaHistoryRequest(this, pathPattern);
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaDiffRequest} that is used to send
+     * {@link CentralDogma#getDiff(String, String, Revision, Revision, Query)} request to the
+     * Central Dogma repository.
+     */
+    public CentralDogmaDiffRequest<?> diff(String path) {
+        requireNonNull(path, "path");
+        return diff(Query.of(QueryType.IDENTITY, path));
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaDiffRequest} that is used to send
+     * {@link CentralDogma#getDiff(String, String, Revision, Revision, Query)} request to the
+     * Central Dogma repository.
+     */
+    public <T> CentralDogmaDiffRequest<T> diff(Query<T> query) {
+        requireNonNull(query, "query");
+        return new CentralDogmaDiffRequest<>(this, query);
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaDiffsRequest} that is used to send
+     * {@link CentralDogma#getDiffs(String, String, Revision, Revision, String)} request to the
+     * Central Dogma repository.
+     * A path pattern is a variant of glob:
+     * <ul>
+     *   <li>{@code "/**"} - find all files recursively</li>
+     *   <li>{@code "*.json"} - find all JSON files recursively</li>
+     *   <li>{@code "/foo/*.json"} - find all JSON files under the directory {@code /foo}</li>
+     *   <li><code>"/&#42;/foo.txt"</code> - find all files named {@code foo.txt} at the second depth level</li>
+     *   <li>{@code "*.json,/bar/*.txt"} - use comma to specify more than one pattern. A file will be matched
+     *                                     if <em>any</em> pattern matches.</li>
+     * </ul>
+     */
+    public CentralDogmaDiffsRequest diffs(String pathPattern) {
```

Comment:
`diffs` is a bit unnatural to me. How about introducing a new query type that can represent multiple files using a path pattern?
For example:
```java
CentralDogmaDiffsRequest file(MultiQuery query){
}
...
CentralDogmaDiffsRequest diff(MultiQuery query){
}
interface MultiQuery {
    static all() {
        return new DefaultMultiQuery("/**");
    }
    // Could represent *.json
    static pattern(String pattern) {    
        ...
    }
    // Could represent *.json, /bar/*.txt
    static patterns(String patrerns...) {
    
    }
}
```